### PR TITLE
Use FBS record for interlibrary loans and reservations where material cannot be found in FBI

### PR DESCRIPTION
**Pull request by kasperg**
#### Link to issue

https://reload.atlassian.net/browse/DDFLSBP-595

#### Description

This PR extends list types used for reservations and loans with the possibility of attaching basic details. This is used when returning data from FBS.

These basic details can then be used as fallback information for physical materials in case these cannot be retrieved from FBI in the material fetch higher-order component.

Without this then the material will not be rendered at all.

One known situation where this occurs is interlibrary loans from the Royal Danish Library.

Add a test showing that this works.

### CODE REVIEW

File: src/apps/loan-list/list/loan-list.test.ts [hunk 1]
```diff
@@ -672,6 +672,84 @@ describe("Loan list", () => {
...
+      body: {
+        data: {}
+      }
+    }).as("work_not_found");
+
+    cy.visit("/iframe.html?path=/story/apps-loan-list--loan-list-entry");
+
+    cy.get(".list-reservation-container")
+      .find(".list-reservation")
+      .get(".list-reservation__header")
```
Comment (kasperg) [inline src/apps/loan-list/list/loan-list.test.ts:738]: It is a bit odd that this test for loans uses selectors which mentions reservation.

However this is the case for this entire test and it is outside the scope of this PR to change this.

File: src/core/utils/helpers/list-mapper.ts [hunk 2]
```diff
@@ -68,7 +68,17 @@ export const mapFBSLoanToLoanType = (list: LoanV2[]): LoanType[] => {
...
-      loanId: loanDetails.loanId
+      loanId: loanDetails.loanId,
+      details: loanDetails.ilBibliographicRecord
+        ? {
+            title: loanDetails.ilBibliographicRecord.title,
+            authors: loanDetails.ilBibliographicRecord.author,
+            authorsShort: loanDetails.ilBibliographicRecord.author,
+            firstAuthor: loanDetails.ilBibliographicRecord.author,
+            year: loanDetails.ilBibliographicRecord.publicationDate,
+            lang: loanDetails.ilBibliographicRecord.language
```
Comment (Dresse) [inline src/core/utils/helpers/list-mapper.ts:79]: Do we not need LSBN for loans?

**Review by Dresse — APPROVED** [review]
Looks good. Just have a single question. Approved 👍

**Review by Dresse — APPROVED** [review]
LGTM 👍

File: src/core/utils/helpers/list-mapper.ts (hunk 2 above)
Comment (kasperg) [reply src/core/utils/helpers/list-mapper.ts:79]: :+1: Fixed.

**Pull request merged**